### ISSUE TITLE
Benchmark command fail due to json deserialize error

### ISSUE THREAD

**Issue by NingLin-P** [paritytech/polkadot-sdk#6306]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

The benchmark command now requires the `genesis-builder-preset` arg to get JSON-encoded `RuntimeGenesisConfig` from the runtime but this value is not used directly, instead it is used to patch the default value of `RuntimeGenesisConfig` and the result is used:
https://github.com/paritytech/polkadot-sdk/blob/01936b346098e4c603f035622dfd24846ac5cb52/substrate/client/chain-spec/src/genesis_config_builder.rs#L153-L155

within `json_patch::merge` if the value is `null` the whole kv pair will be removed from the JSON:
https://github.com/paritytech/polkadot-sdk/blob/01936b346098e4c603f035622dfd24846ac5cb52/substrate/client/chain-spec/src/json_patch.rs#L36-L38

this causes issue for enum that contains `Option` value:
```rust
enum Test {
    V(Option<u32>),
}
```
`Test::V(None)` will be serialized to `{v: null}`, after `json_patch::merge` it became `{ }` and is impossible to deserialize back to the same value

### Steps to reproduce

Add this unit test to `substrate/client/chain-spec/src/json_patch.rs` and it will fail due to deserialize
```rust
#[test]
	fn test_null_value() {
		use serde::{Deserialize, Serialize};

		#[derive(Debug, Serialize, Deserialize)]
		enum Test {
			V(Option<u32>),
		}

		let v1 = serde_json::to_value(Test::V(None)).unwrap();
		let mut v2 = serde_json::to_value(Test::V(None)).unwrap();

		merge(&mut v2, v1);

		let j = serde_json::to_string(&v2).expect("serialize should succeed");

		serde_json::from_slice::<Test>(&j.into_bytes()).expect("deserialize should succeed");
	}
```

**Comment by bkchr** [comment]
CC @michalkucharczyk

**Comment by michalkucharczyk** [comment]
Thanks for reporting and in-depth analysis of the problem. It all sounds good, removal of keys in the base JSON object which is default is indeed not necessary. The fix proposal is here: #6382

**Comment by NingLin-P** [comment]
The fix does works, thanks!
